### PR TITLE
Handle in HF AutoTokenizer with pad_token=None

### PR DESCRIPTION
**Pull request by HuiyingLi**
# What does this PR do ?

In the NML data preprocessing script `scripts/nlp_language_modeling/preprocess_data_for_megatron.py` 
the builder the pad_id thru:
`pad_id=tokenizer.pad_id if hasattr(tokenizer, "pad_id") else 0`

While HF wrapper AutoTokenizer class has `pad_id` as a property, but doesn't actually have pad_token, which will throw None error.
This pr checks if special token exists before encode to get ID of special token.

**Collection**: [Note which collection this PR will affect]

# Changelog 
- Add specific line by line info of high level changes in this PR.

# Usage
* You can potentially add a usage example below

```python
# Add a code snippet demonstrating how to use this 
```

# Jenkins CI
To run Jenkins, a NeMo User with write access must comment `jenkins` on the PR.

# Before your PR is "Ready for review"
**Pre checks**:
- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [ ] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to # (issue)

### CODE REVIEW

**Comment by HuiyingLi** [comment]
Is it possible to reopen the pr?

**Review by aklife97 — APPROVED** [review]
LGTM, thank you!

**Comment by cuichenx** [comment]
jenkins

File: scripts/nlp_language_modeling/preprocess_data_for_megatron.py [hunk 1]
```diff
@@ -309,7 +309,7 @@ def main():
             output_bin_files[key],
             impl=args.dataset_impl,
             chunk_size=args.chunk_size,
-            pad_id=tokenizer.pad_id if hasattr(tokenizer, "pad_id") else 0,
+            pad_id=tokenizer.pad_id if (hasattr(tokenizer, "pad_id") and tokenizer.pad_id is not None) else 0,
```
[the remark this answers is not in the document]
Comment (HuiyingLi) [reply scripts/nlp_language_modeling/preprocess_data_for_megatron.py]: Thank you for the suggestion! Just changed.

**Comment by HuiyingLi** [comment]
@yaoyu-33 Could you review the resolved issue and say jenkins again? Thanks!!

**Comment by yaoyu-33** [comment]
jenkins

**Review by yaoyu-33 — APPROVED** [review]

**Comment by yaoyu-33** [comment]
jenkins

**Pull request merged**